src/main/scala/is/hail/variant/HTSGenotypeView.scala

```diff
@@ -276,4 +278,115 @@ class ArrayGenotypeView(rowType: TStruct) {
     val elementOffset = ArrayGenotypeView.tArrayFloat64.elementOffset(adOffset, length, idx)
     m.loadDouble(elementOffset)
   }
+}
+
+object HardCallView {
+  def apply(rowSignature: TStruct): HardCallView = {
+    rowSignature.fieldType(3).asInstanceOf[TArray].elementType match {
+      case TGenotype => new HardCallTGenotypeView(rowSignature)
+      case _: TStruct => new HardCallStructView(rowSignature, "GT")
+      case t => fatal(s"invalid genotype representation: $t, expect TGenotype or TStruct")
```

Comment:
Should we treat a non-struct like any other type where the field is missing?  The code you wrote below supports that (by setting the element type to null).

Reply:
Won't we never have anything other than a struct?

Reply:
What does it mean to try to read a non-TStruct, non-TGenotype field as a genotype? Are there circumstances where you'd actually want to see e.g. a TInt32 where you expect a genotype and just set the thing to missing instead of throwing an error?